### PR TITLE
feat: new command

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description
This PR adds `new` command for profile preparation.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. For updating Oclif commands documentation use [oclif-dev](https://github.com/oclif/dev-cli#oclif-dev-readme).
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/commands/new.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import { buildProfilePath, buildProviderPath } from '../common/file-structure';
+import { exists, readFile } from '../common/io';
+import type { ILogger } from '../common/log';
+import { OutputStream } from '../common/output-stream';
+import { newProfile } from '../logic/new';
+
+export default class New extends Command {
+  // TODO: add description
+  public static description =
+    'Generates Comlink profile from prepared API documentation. Comlink profile defines interface od API integration. Use name of provider as first argument and description of your use case as second argument';
```
Comment (janhalama) [inline src/commands/new.ts:21]: typo
```suggestion
    'Generates Comlink profile from prepared API documentation. Comlink profile defines interface of API integration. Use name of provider as first argument and description of your use case as second argument';
```

File: src/commands/new.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    if (prompt === undefined) {
+      throw userError(
+        'Missing short description of your use case in natural language. Please provide it as second argument.',
+        1
+      );
+    }
+
+    if (!(await exists(buildProviderPath(providerName)))) {
+      throw userError(
+        `Provider ${providerName} does not exist. Make sure to tun "sf prepare" before running this command.`,
```
Comment (janhalama) [inline src/commands/new.ts:90]:
```suggestion
        `Provider ${providerName} does not exist. Make sure to run "sf prepare" before running this command.`,
```

File: src/logic/new.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      throw Error(
+        `Unexpected response received - unable to parse profile source: ${JSON.stringify(
+          e,
+          null,
+          2
+        )}`
+      );
+    }
+
+    // TODO: validate id format?
```
Comment (janhalama) [inline src/logic/new.ts:51]: Parser validates the profile name. What kind of validation do you mean here?

**Review by janhalama — APPROVED** [review]
LGTM 👍

File: src/logic/new.ts (hunk 3 above)
Comment (Jakub-Vacek) [reply src/logic/new.ts:51]: I was thinking about `.` or `/` separators of scope and name.  But probably it is not needed now.

**Pull request merged**